File: validate_subsequence/validateSubsequence.js
// Psuedo-Code
// Create a pointer for the sequence array
// iteratre over the array of nums
	// check if the current num is the one in the sequence que
		// if so iterate the sequence index
		// if the sequence index is the length of the sequence array, return true
// return false

function isValidSubsequence(array, sequence) {
	let sequenceIdx = 0;
	for (num of array) {
		let targetNum = sequence[sequenceIdx];
		if (num === targetNum) {
			sequenceIdx++;
			if (sequenceIdx === sequence.length) {
				return true;
			}
		}
	}
	return false;
}

File: validate_subsequence/validateSubsequence.py
# Psuedo-Code
# Set pointer for sequence array
# iterate over array of nums
	# if the num at the pointer idx in the sequence arr matchs the arr num
		# Increment pointe 
		# Check if pointers length is the same as the sequence arr
			# if so return true
# return false

def isValidSubsequence(array, sequence):
	sequenceIdx = 0
	for num in array:
		target = sequence[sequenceIdx]
		if num == target:
			sequenceIdx += 1
			if sequenceIdx == len(sequence):
				return True
	return False
